### PR TITLE
feat: Create RTE Toolbar component

### PR DESCRIPTION
**Pull request by dougmacknz**
# Objective
Create a UI for the toolbar and toolbar section for the Rich Text Editor

# Motivation and Context
- this part of the inner UI components that support the rich text editor

### CODE REVIEW

File: packages/rich-text-editor/docs/Toolbar.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  },
+  decorators: [withDesign],
+}
+
+export const DefaultStory = (args: ToolbarProps) => (
+  <Toolbar {...args}>
+    <ToolbarSection>
+      <ToolbarItem>
+        <ToggleIconButton label="Bold" icon={boldIcon} />
+      </ToolbarItem>
```
Comment (mcwinter07) [inline packages/rich-text-editor/docs/Toolbar.stories.tsx:43]: I've update the styles so we don't actually need the ToolbarItem wrapper assuming that the first child is an inline block or flex. I've left it in for now until we know we don't need it

File: packages/rich-text-editor/src/RichTextEditor/components/Toolbar/Toolbar.scss [hunk 2]
```diff
@@ -0,0 +1,7 @@
+@import "@kaizen/design-tokens/sass/color";
+@import "@kaizen/design-tokens/sass/spacing";
+
+.toolbar {
+  display: block;
+  border-bottom: 2px solid $color-gray-600;
```
Comment (mcwinter07) [inline packages/rich-text-editor/src/RichTextEditor/components/Toolbar/Toolbar.scss:6]: This border could belong here or in the actual text input wrapper as a border top. Either way one consideration is how we handle the colour on the error styles - I've left it out for now till we have all the components

File: packages/rich-text-editor/src/RichTextEditor/components/Toolbar/Toolbar.tsx [hunk 3]
```diff
@@ -0,0 +1,104 @@
+import React, { Children, useState, useContext, useRef, ReactNode } from "react"
+import classnames from "classnames"
+import { ToolbarSectionProps } from "../ToolbarSection"
+import { ToolbarItem, ToolbarItemProps } from "../ToolbarItem"
+import styles from "./Toolbar.scss"
+
+export interface ToolbarProps extends React.HTMLAttributes<HTMLDivElement> {
+  children: React.ReactNode
+  /*
+   * Connect the Toolbar with the editable content area though its ID
```
Comment (dougmacknz) [inline packages/rich-text-editor/src/RichTextEditor/components/Toolbar/Toolbar.tsx:10]:
```suggestion
   * Connect the Toolbar with the editable content area through its ID
```

File: packages/rich-text-editor/src/RichTextEditor/components/Toolbar/Toolbar.tsx [hunk 4]
```diff
@@ -0,0 +1,104 @@
+import React, { Children, useState, useContext, useRef, ReactNode } from "react"
+import classnames from "classnames"
+import { ToolbarSectionProps } from "../ToolbarSection"
+import { ToolbarItem, ToolbarItemProps } from "../ToolbarItem"
+import styles from "./Toolbar.scss"
+
+export interface ToolbarProps extends React.HTMLAttributes<HTMLDivElement> {
```
Comment (dougmacknz) [inline packages/rich-text-editor/src/RichTextEditor/components/Toolbar/Toolbar.tsx:7]: We should do the `classNameOverride` thing here, and we can use the `@kaizen/component-base` stuff that Cass built. E.g. https://github.com/cultureamp/kaizen-design-system/pull/2632/files#diff-6da5f01f257e37406d113e4bf53b6547954456add88abfd0b8254790fda90515R3

File: packages/rich-text-editor/src/RichTextEditor/components/ToolbarItem/ToolbarItem.tsx [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import React from "react"
+import styles from "./ToolbarItem.scss"
+
+export interface ToolbarItemProps extends React.HTMLAttributes<HTMLDivElement> {
+  children: React.ReactNode
+}
+
+export const ToolbarItem: React.VFC<ToolbarItemProps> = props => (
```
Comment (dougmacknz) [inline packages/rich-text-editor/src/RichTextEditor/components/ToolbarItem/ToolbarItem.tsx:8]: Looks like you forgot to remove all of the ToolbarItem stuff given you're not using it anymore

**Review by dougmacknz — COMMENTED** [review]
Couple of minor things and this is looking good to go 👏

**Review by mcwinter07 — APPROVED** [review]

**Pull request merged**